<launch>

    <!-- Transformation Configuration ... Setting Up the Relationships Between Coordinate Frames --> 
    <node pkg="tf2_ros" type="static_transform_publisher" name="base_link_to_laser" args="0.45 0 0.0 0 0 0 base_link laser" />

    <node name="rplidarNode"          pkg="rplidar_ros"  type="rplidarNode" output="screen">
        <param name="serial_port"         type="string" value="/dev/ttyUSB0"/>
        <param name="serial_baudrate"     type="int"    value="115200"/><!--A1/A2 -->
        <!--param name="serial_baudrate"     type="int"    value="256000"--><!--A3 -->
        <param name="frame_id"            type="string" value="laser"/>
        <param name="inverted"            type="bool"   value="false"/>
        <param name="angle_compensate"    type="bool"   value="true"/>
    </node>

    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
        <rosparam file="$(find dumb_wall_follower)/param/costmap_common_params.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find dumb_wall_follower)/param/costmap_common_params.yaml" command="load" ns="local_costmap" />
        <rosparam file="$(find dumb_wall_follower)/param/local_costmap_params.yaml" command="load" ns="local_costmap" />
        <rosparam file="$(find dumb_wall_follower)/param/global_costmap_params.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find dumb_wall_follower)/param/base_local_planner_params.yaml" command="load" />
    </node>

    <!-- <node pkg="dumb_wall_follower" name="laser_scan_listener" type="main_node.py" output="screen">
    </node> -->

    <node pkg="dumb_wall_follower" name="move_client_node" type="move_client_node.py" output="screen">
    </node>
    
    

</launch>